hooks/boot/04-services.rb

```diff
@@ -0,0 +1,36 @@
+require 'open3'
+
+module ServicesHookContextExtension
+  ALL_POSSIBLE_SERVICES = [
+    'httpd.service', 'apache.service', # RH vs Debian
+    'postgresql*',
+    'foreman*',
+    'dynflow*',
+    '*mongo*',
+    '*redis*',
+    'pulp*',
+  ]
+
+  def start_services(services)
+    raise "Services must be specified" if services.empty?
```

Comment:
Are you expecting Kafo to catch this and deal with it? Otherwise I think we can to use `fail`

Reply:
Passing an empty array is incorrect API use and the goal of this is to slap the developer. Users should not see this. The alternative is to just return.

Reply:
This was mostly around the nuance of raise vs fail in Ruby -- https://gist.github.com/Integralist/8058ac54088408d9aadc

Reply:
TIL. Thanks!